cloud-agent/service/server/src/main/scala/org/hyperledger/identus/agent/server/jobs/PresentBackgroundJobs.scala

```diff
@@ -66,152 +78,24 @@ object PresentBackgroundJobs extends BackgroundJobsHelper {
-      proverDID <- ZIO
-        .fromEither(PrismDID.fromString(vcSubjectId))
-        .mapError(e =>
-          PresentationError
-            .UnexpectedError(
-              s"One of the credential(s) subject is not a valid Prism DID: ${vcSubjectId}"
-            )
-        )
-      longFormPrismDID <- getLongForm(proverDID, true)
-      jwtIssuer <- createJwtIssuer(longFormPrismDID, VerificationRelationship.Authentication)
-    } yield jwtIssuer
-
-  // Holder / Prover Get the Holder/Prover PrismDID from the IssuedCredential
-  // SDJWT Only currrently When holder accepts offer he provides the subjectDid and optional keyId which is used for key binding
-  private def getHolderPrivateKeyFromCredentials(
-      presentationId: DidCommID,
-      credentialsToUse: Seq[String]
-  ) =
-    for {
-      credentialService <- ZIO.service[CredentialService]
-      // Choose first credential from the list to detect the subject DID to be used in Presentation.
-      // Holder binding check implies that any credential record can be chosen to detect the DID to use in VP.
-      credentialRecordId <- ZIO
-        .fromOption(credentialsToUse.headOption)
-        .mapError(_ =>
-          PresentationError.UnexpectedError(s"No credential found in the Presentation record: $presentationId")
-        )
-      credentialRecordUuid <- ZIO
-        .attempt(DidCommID(credentialRecordId))
-        .mapError(_ => PresentationError.UnexpectedError(s"$credentialRecordId is not a valid DidCommID"))
-      credentialRecord <- credentialService
-        .findById(credentialRecordUuid)
-        .someOrFail(CredentialServiceError.RecordNotFound(credentialRecordUuid))
-      vcSubjectId <- ZIO
-        .fromOption(credentialRecord.subjectId)
-        .orDieWith(_ => RuntimeException(s"VC SubjectId not found in credential record: $credentialRecordUuid"))
-
-      proverDID <- ZIO
-        .fromEither(PrismDID.fromString(vcSubjectId))
-        .mapError(e =>
-          PresentationError
-            .UnexpectedError(
-              s"One of the credential(s) subject is not a valid Prism DID: ${vcSubjectId}"
-            )
-        )
-      longFormPrismDID <- getLongForm(proverDID, true)
-
-      optionalHolderPrivateKey <- credentialRecord.keyId match
-        case Some(keyId) =>
-          findHolderEd25519SigningKey(
-            longFormPrismDID,
-            VerificationRelationship.Authentication,
-            keyId
-          ).map(ed25519keyPair => Option(HolderPrivateKey(ed25519keyPair.privateKey)))
-        case None => ZIO.succeed(None)
-
-    } yield optionalHolderPrivateKey
+  private def performPresentProofExchange(record: PresentationRecord): URIO[RESOURCES, Unit] =
+    aux(record)
+      .tapError({
+        (error: PresentationError | DIDSecretStorageError | BackgroundJobError | CredentialServiceError |
+          CastorDIDResolutionError | GetManagedDIDError | TransportError) =>
+          ZIO.logErrorCause(
```

Comment:
Seems we have error log in several places.
Just make sure we are not double logging